src/st/no0/e_diplocephalus_2.c

```diff
@@ -0,0 +1,326 @@
+            tempEntity = tempEntity->ext.diplocephalus.unk90;
+        }
+        i = 0;
+        while (tempEntity->ext.diplocephalus.unk90 != NULL) {
+            tempEntity->ext.diplocephalus.unk84 = temp_a2 + temp_s6 * i;
+            xOffset = (rcos(tempEntity->ext.diplocephalus.unk84) << 3) << 4;
+            yOffset = (rsin(tempEntity->ext.diplocephalus.unk84) << 3) << 4;
+            if (self->facingLeft) {
+                tempEntity->ext.diplocephalus.unk90->posX.val =
+                    tempEntity->posX.val - xOffset;
+            } else {
+                tempEntity->ext.diplocephalus.unk90->posX.val =
+                    tempEntity->posX.val + xOffset;
+            }
+            tempEntity->ext.diplocephalus.unk90->posY.val =
+                tempEntity->posY.val - yOffset;
+            i++;
+            tempEntity->rotZ = -tempEntity->ext.diplocephalus.unk84;
+            tempEntity = tempEntity->ext.diplocephalus.unk90;
+        }
+        self->ext.diplocephalus.entity->posX.i.hi += 1;
+        self->ext.diplocephalus.entity->posY.i.hi -= 2;
+        self->ext.diplocephalus.velocityY =
+            self->ext.diplocephalus.parent->velocityY;
+        self->ext.diplocephalus.velocityX =
+            self->ext.diplocephalus.parent->velocityX;
+        self->ext.diplocephalus.unk9C++;
+        if (self->ext.diplocephalus.unk9C > 0x20) {
+            tempEntity = self->ext.diplocephalus.unk90;
+            for (i = 0; i < self->ext.diplocephalus.unk9D; i++) {
+                tempEntity = tempEntity->ext.diplocephalus.unk90;
+            }
+            tempEntity->animFrameDuration = 0;
+            tempEntity->animFrameIdx = 0;
+            tempEntity->step = 4;
+            self->ext.diplocephalus.unk9C = 0;
+            self->ext.diplocephalus.unk9D++;
+        }
+        if (self->ext.diplocephalus.unk9D > 3) {
+            self->ext.diplocephalus.unk9D = 0;
+        }
+        break;
+    case 4:
+        if (AnimateEntity(D_us_80181E30, self) == 0) {
+            self->animFrameDuration = 0;
+            self->animFrameIdx = 0;
+            self->step = 3;
+        }
+        break;
+    case 5:
+        AnimateEntity(D_us_80181DBC, self);
+        if (self->ext.diplocephalus.parent->ext.diplocephalus.unkAD != 0) {
+            PlaySfxPositional(0x777);
+            self->animFrameDuration = 0;
+            self->animFrameIdx = 0;
+            self->step = 6;
+            return;
+        }
+        if (!(Random() & 0x7F)) {
+            self->animFrameDuration = 0;
+            self->animFrameIdx = 0;
+            self->params = 1;
+            self->step = 11;
+        }
+        break;
+    case 6:
+        if (AnimateEntity(D_us_80181DDC, self) == 0) {
+            self->animFrameDuration = 0;
+            self->animFrameIdx = 0;
+            self->step--;
+        }
+        break;
+    case 7:
+        if (AnimateEntity(D_us_80181DEC, self) == 0) {
+            tempEntity = AllocEntity(&D_8007A958, &D_8007A958 + 32);
```

Comment:
I think this is `&g_Entities[160], &g_Entities[192]` unless I miscounted

`g_Entities` @ 0x800733D8
0x80181DEC - 0x800733D8 = 0x7580
0x7580 / 0xBC = 0xA0 / 160